libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+    /**
+     * Wipes all components currently being logged. Used only by tests.
+     */
+    public synchronized static void flushComponents() {
+        LOGGERS = null;
+    }
+
+    private SalesforceLogger(String componentName, Context context) {
+        this.context = context;
+        if (isDebugMode()) {
+            logLevel = Level.DEBUG;
+        } else {
+            logLevel = Level.ERROR;
+        }
+        try {
+            fileLogger = new FileLogger(context, componentName);
+        } catch (IOException e) {
+            Log.e(TAG, "Couldn't create file logger", e);
+        }
+    }
+
+    private boolean isDebugMode() {
+        boolean debugMode = true;
+        try {
+            final PackageManager pm = context.getPackageManager();
+            if (pm != null) {
+                final PackageInfo pi = pm.getPackageInfo(context.getPackageName(), 0);
+                if (pi != null) {
+                    final ApplicationInfo ai = pi.applicationInfo;
+                    if (ai != null && ((ai.flags & ApplicationInfo.FLAG_DEBUGGABLE) == 0)) {
+                        debugMode = false;
+                    }
+                }
+            }
+        } catch (PackageManager.NameNotFoundException e) {
+            debugMode = true;
+        }
+        return debugMode;
+    }
+
+    /**
+     * Returns the instance of FileLogger associated with this component.
+     *
+     * @return FileLogger instance.
+     */
+    public FileLogger getFileLogger() {
+        return fileLogger;
+    }
+
+    /**
+     * Returns the log level currently being used.
+     *
+     * @return Log level.
+     */
+    public Level getLogLevel() {
+        return logLevel;
+    }
+
+    /**
+     * Sets the log level to be used.
+     *
+     * @param level Log level.
+     */
+    public synchronized void setLogLevel(Level level) {
+        this.logLevel = level;
+    }
+
+    /**
+     * Disables file logging.
+     */
+    public synchronized void disableFileLogging() {
+        if (fileLogger != null) {
+            fileLogger.setMaxSize(0);
+        }
+    }
+
+    /**
+     * Enables file logging.
+     *
+     * @param maxSize Maximum number of log lines allowed to be stored at a time.
+     */
+    public synchronized void enableFileLogging(int maxSize) {
+        if (fileLogger != null) {
+            fileLogger.setMaxSize(maxSize);
+        }
+    }
+
+    /**
+     * Returns if file logging is enabled or not.
+     *
+     * @return True - if enabled, False - otherwise.
+     */
+    public boolean isFileLoggingEnabled() {
+        int maxSize = 0;
+        if (fileLogger != null) {
+            maxSize = fileLogger.getMaxSize();
+        }
+        return (maxSize > 0);
+    }
+
+    /**
+     * Logs the message passed in at the level specified.
+     *
+     * @param level Log level.
+     * @param tag Log tag.
+     * @param message Log message.
+     */
+    public void log(Level level, String tag, String message) {
```

Comment:
Simple `log` API. Logs to console using stock `Android` logger and then writes the log line to file as well.